nucliadb/nucliadb/reader/api/v1/services.py

```diff
@@ -217,17 +217,23 @@ async def get_labelset(request: Request, kbid: str, labelset: str) -> LabelSet:
 @requires(NucliaDBRoles.READER)
 @version(1)
 async def get_custom_synonyms(request: Request, kbid: str):
-    ingest = get_ingest()
-    pbrequest = KnowledgeBoxID(uuid=kbid)
-    pbresponse: GetSynonymsResponse = await ingest.GetSynonyms(pbrequest)  # type: ignore
-    if pbresponse.status.status == OpStatusWriter.Status.OK:
-        return KnowledgeBoxSynonyms.from_message(pbresponse.synonyms)
-    elif pbresponse.status.status == OpStatusWriter.Status.NOTFOUND:
-        raise HTTPException(status_code=404, detail="Knowledge Box does not exist")
-    elif pbresponse.status.status == OpStatusWriter.Status.ERROR:
-        raise HTTPException(
-            status_code=500, detail="Error getting synonyms of a Knowledge box"
-        )
+    async with datamanagers.with_transaction(read_only=True) as txn:
+        if not datamanagers.kb.exists_kb(txn, kbid=kbid):
+            raise HTTPException(status_code=404, detail="Knowledge Box does not exist")
+
+        try:
+            synonyms = await datamanagers.synonyms.get_kb_synonyms(txn, kbid=kbid)
+        except Exception as exc:
+            errors.capture_exception(exc)
+            logger.exception("Errors getting synonyms")
+            raise HTTPException(
+                status_code=500, detail="Error getting synonyms of a Knowledge box"
+            )
```

Comment:
You can skip the generic error handling, there is a fastapi global handler already setup